tests/syntax_tests/data/parsing/errors/expressions/expected/array.res.txt

```diff
@@ -8,4 +8,4 @@
 
   Did you forget a `]` here?
 
-let xs = x.map (Function$ (fun [arity:1]key -> [|key;(predicates.(key))|]))
\ No newline at end of file
+let xs = x.map (fun [arity:1]key -> [|key;(predicates.(key))|])
```

Comment:
parse tests should only differ by removing `Function$`.

Reply:
Sometimes only the `Function$` is removed, but sometimes also the `fun`?

Reply:
That's the ocaml pretty printer.
Two nested fun defs without a constructor such as Function$ in the middle are squashed into a single `fun`.

Reply:
Yes, but I mean like the one I commented on below:

```rescript
let f = Function$ (fun [arity:2]g -> fun h -> ((a + 3; b)[@res.braces ]))
```

->

```rescript
let f [arity:2]g h = ((a + 3; b)[@res.braces ])
```

where there is no `fun` at all left.

Reply:
Yes that's ocaml printing

Reply:
We can revisit this printing too, but it's only for debugging.
The alternative would be use the real AST as a tree but that's super verbose.
Perhaps one could use an explicit ReScript-style printing but that corresponds 1 to 1 to the AST without magic.